pkg/tests/end_to_end_tests/trace_query_integration_test.go

```diff
@@ -0,0 +1,396 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package end_to_end_tests
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/golang/snappy"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/stretchr/testify/require"
+	jaegerquery "github.com/timescale/promscale/pkg/jaeger/query"
+	ingstr "github.com/timescale/promscale/pkg/pgmodel/ingestor"
+	"github.com/timescale/promscale/pkg/pgxconn"
+
+	jaegerQueryApp "github.com/jaegertracing/jaeger/cmd/query/app"
+	jaegerQueryService "github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+	jaegerJSONModel "github.com/jaegertracing/jaeger/model/json"
+)
+
+type traceQuery struct {
+	name    string
+	start   int64
+	end     int64
+	service string
+	traceID [16]byte // For testing getTrace API.
+}
+
+var traceQueryCases = []traceQuery{
```

Comment:
we need a comment above this about what command to run to regenerate the response. I see not instructions about how to use the gen_responses tag or how to regenerate the dataset.

Reply:
I wrote them on top of `TestGenerateJaegerAPIResponses` but will add a short desc here too.